services/api.py

```diff
@@ -830,19 +830,31 @@ def get_category_ids(categories, category):
             queryset = queryset.filter(Q(services__in=service_ids)
                                        | Q(service_nodes__in=service_nodes_by_ancestors(servicenode_ids))).distinct()
 
+        def get_services_by_servicenodes(servicenode_ids):
+            ids = []
+            servicenodes = ServiceNode.objects.filter(id__in=service_nodes_by_ancestors(servicenode_ids))
+            for node in servicenodes:
+                servicelist = node.related_services.all()
+                ids.extend([service.id for service in servicelist])
+            return ids
+
+        # TODO: add validation
         if 'period' in filters:
             period = filters.get('period', None)
             services = None
             if 'service' in filters:
                 services = filters.get('service').split(',')
```

Comment:
Currently this duplicates code found elsewhere inside this method.

We should refactor so that the filter query parameter parsing is in one place, and the results are shared by this period filter along with the actual service/service_node/category filters. You might also put some of the DB queries in one place.